### PR TITLE
rootless: avoid hang on failed slirp4netns

### PR DESCRIPTION
**Pull request by giuseppe**
If for any reason slirp4netns fails at startup, podman waits
indefinitely.  Check every second if the process is still running so
that we avoid to hang.

I've noticed this issue while I was playing with slirp4netns and setting
an option that is not supported.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: libpod/networking_linux.go [hunk 1]
```diff
@@ -136,10 +137,30 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
...
 		return errors.Wrapf(err, "failed to start process")
 	}
+	defer cmd.Process.Release()
 
 	b := make([]byte, 16)
-	if _, err := syncR.Read(b); err != nil {
-		return errors.Wrapf(err, "failed to read from sync pipe")
+	for {
+		if err := syncR.SetDeadline(time.Now().Add(1 * time.Second)); err != nil {
+			return errors.Wrapf(err, "error setting pipe timeout")
```
Comment (rhatdan) [inline libpod/networking_linux.go]: Your error messages should mention that the problem is with slirp4netns hanging.

Comment (giuseppe) [reply libpod/networking_linux.go]: I changed the error messages to mention slirp4netns

**Comment by rhatdan** [comment]
LGTM
/approve

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons.

**Pull request merged**